diff --git a/gdb/valprint.c b/gdb/valprint.c
index ce1ac753aa4..b1560f80f24 100644
--- a/gdb/valprint.c
+++ b/gdb/valprint.c
@@ -2234,15 +2234,15 @@ print_wchar (gdb_wint_t w, const gdb_byte *orig,
 						  byte_order);
 		  /* If the value fits in 3 octal digits, print it that
 		     way.  Otherwise, print it as a hex escape.  */
-		  if (value <= 0777)
+		  // if (value <= 0777)
+		  //   {
+		  //     xsnprintf (octal, sizeof (octal), "\\%.3o",
+				 // (int) (value & 0777));
+		  //     *need_escapep = false;
+		  //   }
+		  // else
 		    {
-		      xsnprintf (octal, sizeof (octal), "\\%.3o",
-				 (int) (value & 0777));
-		      *need_escapep = false;
-		    }
-		  else
-		    {
-		      xsnprintf (octal, sizeof (octal), "\\x%lx", (long) value);
+		      xsnprintf (octal, sizeof (octal), "\\x%02lx", (long) value);
 		      /* A hex escape might require the next character
 			 to be escaped, because, unlike with octal,
 			 hex escapes have no length limit.  */
@@ -2255,7 +2255,7 @@ print_wchar (gdb_wint_t w, const gdb_byte *orig,
 		{
 		  char octal[5];
 
-		  xsnprintf (octal, sizeof (octal), "\\%.3o", orig[i] & 0xff);
+		  xsnprintf (octal, sizeof (octal), "\\%02x", orig[i] & 0xff);
 		  *need_escapep = false;
 		  append_string_as_wide (octal, output);
 		  ++i;