diff --git a/gdb/valprint.c b/gdb/valprint.c
index ce1ac753aa4..469d68b1f7f 100644
--- a/gdb/valprint.c
+++ b/gdb/valprint.c
@@ -2232,22 +2232,12 @@ print_wchar (gdb_wint_t w, const gdb_byte *orig,
 
 		  value = extract_unsigned_integer (&orig[i], width,
 						  byte_order);
-		  /* If the value fits in 3 octal digits, print it that
-		     way.  Otherwise, print it as a hex escape.  */
-		  if (value <= 0777)
-		    {
-		      xsnprintf (octal, sizeof (octal), "\\%.3o",
-				 (int) (value & 0777));
-		      *need_escapep = false;
-		    }
-		  else
-		    {
-		      xsnprintf (octal, sizeof (octal), "\\x%lx", (long) value);
-		      /* A hex escape might require the next character
-			 to be escaped, because, unlike with octal,
-			 hex escapes have no length limit.  */
-		      *need_escapep = true;
-		    }
+		  /* Always emit a hex escape, never octal.  */
+		  xsnprintf (octal, sizeof (octal), "\\x%02lx", (long) value);
+		  /* A hex escape might require the next character
+		     to be escaped, because, unlike with octal,
+		     hex escapes have no length limit.  */
+		  *need_escapep = true;
 		  append_string_as_wide (octal, output);
 		}
 	      /* If we somehow have extra bytes, print them now.  */
@@ -2255,8 +2245,8 @@ print_wchar (gdb_wint_t w, const gdb_byte *orig,
 		{
 		  char octal[5];
 
-		  xsnprintf (octal, sizeof (octal), "\\%.3o", orig[i] & 0xff);
-		  *need_escapep = false;
+		  xsnprintf (octal, sizeof (octal), "\\x%02x", orig[i] & 0xff);
+		  *need_escapep = true;
 		  append_string_as_wide (octal, output);
 		  ++i;
 		}
